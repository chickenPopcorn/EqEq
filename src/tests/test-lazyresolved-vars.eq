SomeCtx = {
  a = { b + 40; }
  b = { 2 ^ c + 5; }
  c = { 5; }
  f = { 1/3; } /* never used */
}

SomeCtx:find a {
  d = 100; /* never used */
  print("a = %.0f (when b = %.0f)\n", a, b);

  b = 2; /* now: a depends on b and b is independent */
  print("a = %.0f (when b = %.0f)\n", a, b);
}
