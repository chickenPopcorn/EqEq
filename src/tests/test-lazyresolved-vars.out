a = 42 (when b = 2)
